shared/challenge_builder.py

```diff
@@ -2,77 +2,55 @@
 from dataclasses import dataclass, field
 from typing import List, Dict
 import requests
+import geojson
+from turfpy.measurement import distance, bbox, centroid
 
 @dataclass
 class Geometry:
```

Comment:
I very much tried to avoid it, but it just makes things more simple so...

Please only keep the code of the `fromOverpassElement(element, GeomType=None):` function and make it return direct [geojson objects](https://pypi.org/project/geojson/#geojson-objects).
Example:
```py
        if GeomType == "Point":
            if 'geometry' in element:
                return geojson.Point(element['geometry']['coordinates'])
```
Remove the entire Geometry class with all it other functions.


`GeoFeature` can then extend the `Feature` class from https://pypi.org/project/geojson/ wherefore the only thing you have to implement is

```
    @classmethod
    def withId(cls, osmType, osmId, geometry, properties):
        properties["@id"] = f"{osmType}/{osmId}"
```
because the rest should already be done by the `Feature` class.


Finally we adjust the methods of the `Overpass` class

- rename: `queryElements` ->` queryElementsRaw`
- add: `queryElementsAsGeoJSON()`  - this internally calls `queryElementsRaw` and then contains all the code from `Geometry.fromOverpassElement` so that it returns a list of geojson objects

_____


If you want to get the centroid in any of your challenge scripts you can call `centroid(geojson_obj)` 
The functions `convertPolygonToClosedString` and `convertClosedStringToPolygon` are not as readable anymore but still one-liners:
-  `convertPolygonToClosedString`  -> `geojson.LineString(polygon.coordinates[0])`
- `convertClosedStringToPolygon` ->`geojson.Polygon([lineString.coordinates])`